### PR TITLE
experimental compute node lz4 compression

### PR DESCRIPTION
**Pull request by michel2323**
This creates a new PPLIST option LZ4COMMCOMP. That way the compression happens on the compute nodes instead of the IO nodes. 

It only implements the write and should be considered highly experimental. Not sure it belongs into any "stable" branch. 

The read in nek is not symmetric at all. The io node collects all data in one message from all nodes and writes to disk. However, the io nodes reads the data and sends ELEMENT BY ELEMENT to each node. Not sure why this is so and this would be totally different with compressed data as the io node has no notion of elements in the compressed data it has just read.

tl;dr highly experimental write, implementation of read will take a while, please test whether it's worthwhile.

### CODE REVIEW

**Comment by michel2323** [comment]
Read is implemented. Ready for being merged and tested.
